src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -48,6 +48,11 @@ public Task<FeatureDefinition> GetFeatureDefinitionAsync(string featureName)
                 throw new ArgumentNullException(nameof(featureName));
             }
 
+            if (featureName.Contains(ConfigurationPath.KeyDelimiter))
+            {
+                throw new ArgumentException($"The value '{ConfigurationPath.KeyDelimiter}' is not allowed in the feature name.", nameof(featureName));
```

Comment:
If our intent is truly to block ':', which I believe it is, then I think we should define the ':' ourselves. Right now the disallowed char is dependent upon the value of ConfigurationPath.KeyDelimiter. Even if that were to change, we wouldn't necessarily be ready to change our implementation here.